web/components/space-homepage/SpaceLandingPage.tsx

```diff
@@ -110,16 +116,81 @@ export function SpaceLandingPage() {
     [profileListingData?.profile_listing]
   );
 
+  
   const filteredProfileListings = useMemo(() => {
     if (searchQuery === "")
       return shuffleProfiles(allProfileListings, getDayOfYear(new Date()));
 
     const searchQueryLower = searchQuery.toLowerCase();
     const fuse = new Fuse(allProfileListings, FUSE_OPTIONS);
-
     return fuse.search(searchQueryLower).map((result) => result.item);
   }, [allProfileListings, searchQuery]);
 
+  // loop through filteredProfileListings!
+  
+  // idToProfileScore (user's id : how "filled out" their profile is!)
+  const ids = [];
+  const profileScores = [];
+  const idsToProfileScores = new Map();
```

Comment:
Here you're calculating `idsToProfileScores` based on `filteredProfileListings`.

However, because the code is in the main body of the function, this code will run on every render. This is not ideal--you only want to recalculate it when `filteredProfileListings` changes.

Look into React.useMemo to accomplish this: https://react.dev/reference/react/useMemo